core/commands/generate_command.rb

```diff
@@ -16,10 +17,14 @@ def execute
     check_result = setup_command
     return check_result unless check_result == SUCCESS_RESULT
 
-    if @template.template_type == :vagrant
+    case @template.template_type
+    when :vagrant
       generator = VagrantConfigurationGenerator.new(@args, @env, @ui)
       generator.execute(@args.first, @env.override)
-    else
+    when :aws
```

Comment:
Please discuss with me: how difficult will it be to provide Google Cloud as a back-end into Terraform classes. Currently the use of :aws to distinguish Vagrant vs Terraform vs Docker seems odd to me.